packages/react-components/src/components/ProductSwitcher/hooks/useProductSwitcher.ts

```diff
@@ -0,0 +1,46 @@
+import { prodRedirectData, ProductSwitcherProducts } from '../constants';
+import {
+  createProductOption,
+  getRedirectDataByEnv,
+  isVisibleProduct,
+} from '../helpers';
+import { ProductName, ProductOption, ProductSubscription } from '../types';
+
+export interface IProductSwitcherHook {
+  env?: 'labs' | 'staging' | 'prod';
+  installedProducts: {
+    product: ProductName;
+  }[];
+  organizationId: string;
+  subscriptions: ProductSubscription;
+}
+
+export const useProductSwitcher = ({
+  env,
+  installedProducts,
+  subscriptions,
+  organizationId,
+}: IProductSwitcherHook): { products: ProductOption[] } => {
+  if (!installedProducts || !subscriptions || !organizationId) {
+    throw new Error('Missing required parameters');
+  }
+  const redirectData = env ? getRedirectDataByEnv(env) : prodRedirectData;
+
+  return {
+    products: ProductSwitcherProducts.reduce((acc, product) => {
```

Comment:
by doing that you could remove casting at the end of the reduce
```suggestion
    products: ProductSwitcherProducts.reduce<ProductOption[]>((acc, product) => {
```